mlir/compiler-course/kologkin_g_lab_4_var_5/ReplaceCeilWithFloorNegPlugin.cpp

```diff
@@ -0,0 +1,76 @@
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace mlir;
+
+namespace {
+
+void replaceCeilWithNegFloor(ModuleOp module) {
+
+  for (auto func : module.getOps<mlir::func::FuncOp>()) {
+
+    func.walk([&](Operation *op) {
+      if (auto ceilOp = dyn_cast<math::CeilOp>(op)) {
+        Location loc = ceilOp->getLoc();
+        Value input = ceilOp->getOperand(0);
+        auto type = input.getType();
+
+        if (auto vectorType = type.dyn_cast<VectorType>()) {
+          if (!vectorType.getElementType().isa<FloatType>())
+            return;
+        } else if (!type.isa<FloatType>()) {
+          return;
+        }
+
+        OpBuilder builder(ceilOp);
+        builder.setInsertionPoint(ceilOp);
```

Comment:
```suggestion
```
Since you create `builder` object using this operation, this `setInsertionPoint` does nothing